client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -54,15 +54,26 @@ const PatternAssembler: Step = ( { navigation } ) => {
 		return pageTemplate;
 	};
 
+	let incrementIndex = 0;
```

Comment:
I'd prefer to use `uuid` here. There might be a potential issue If we use the `incrementIndex`. For example, the `incrementIndex` would be reset once `PatternAssembler` re-render and then we could get the duplicate key if the pattern is selected multiple times

Reply:
My intention to use the `incrementIndex` is to avoid duplicate key and persist the pattern key in the list during re-render for the animation to work, it should be placed outside of the functional component, not sure why I ended up putting it inside :zany\_face:

```javascript
let incrementIndex = 0;

const PatternAssembler: Step = ( { navigation } ) => {                                                            
```

Though I understand using uuid would eliminate the issue, I just wanna ask your thoughts about using incrementIndex like this, I tested the code and it will increase the value everytime and no duplication index is made 🤗

Reply:
I'm okay with the `incrementIndex` for simplicity. But could we keep it inside the component scope? We can use `useRef` to avoid the value being reset after re-rendering. For example,

```js
const incrementIndexRef = useRef( 0 );

incrementIndexRef.current++;
```

I'd prefer to keep the value starting from 0 but it might not be a big issue in this case.

Reply:
`useRef` is good fit for this!

Reply:
`useRef` is good fit for this!